projects/js-packages/shared-extension-utils/src/get-block-icon-from-metadata.native.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable jsdoc/no-undefined-types */
+import { SvgXml } from 'react-native-svg';
```

Comment:
@SiobhyB My only concern about importing `react-native-svg` is that it's not a dependency in the Jetpack repository. In the future, if we run unit tests for the native version in this repository, this might be an issue as this package won't be exposed and tests might fail.

One option we could explore is to export `SvgXml` as part of the primitives of SVG and the Gutenberg components package.

* https://github.com/WordPress/gutenberg/blob/trunk/packages/primitives/src/svg/index.native.js

This way we could import `SvgXml` here via the `@wordpress/primitives`. Strictly speaking, `SvgXml` is not a primitive of SVG, but we could consider it as this due to be the only solution for processing a SVG coming from string in the native version.

Reply:
> My only concern about importing react-native-svg is that it's not a dependency in the Jetpack repository. In the future, if we run unit tests for the native version in this repository, this might be an issue as this package won't be exposed and tests might fail.

@fluiddot, thank you for sharing, I hadn't considered that concern, and can see how it could be viewed as a blocker now.

I like your suggestion of exporting `SvgXml` via `@wordpress/primitives`, as I do think the function could be considered relevant to that package. I've gone ahead to implement that in this PR at https://github.com/Automattic/jetpack/pull/33890/commits/647795fb67a69ed5eb93b5b497c28f26be6804dc and in Gutenberg at https://github.com/WordPress/gutenberg/pull/55742/commits/49f94c4acc20135c111c529583f00871d166f84e. Looking forward to hearing what you think!